lib/browser/http.ts

```diff
@@ -173,15 +175,15 @@ export class HttpRequest {
 export class HttpClientConnection extends BufferedEventEmitter {
     readonly axios: any;
     constructor(
+        protected boostrap: ClientBootstrap,
         host_name: string,
         port: number,
-        scheme?: string,
+        protected socket_options: SocketOptions,
+        protected tls_opts?: TlsConnectionOptions,
         proxy_options?: HttpProxyOptions,
     ) {
         super();
-        if (!scheme) {
-            scheme = (port == 443) ? 'https' : 'http'
-        }
+        let scheme = (tls_opts || port == 443) ? 'https' : 'http'
```

Comment:
what if I'm using a non-standard tls port? like 8883 or 8443?

Reply:
yeah, I guess we could just do port != 80? But then, what about 8080? Maybe we shouldn't try to interpret port?